### PR TITLE
merge pyniva to master

### PR DESCRIPTION
**Pull request by ZofiaRudjord**
I have tested pyniva with upgraded pandas version locally.

### CODE REVIEW

File: pyniva/get_data.py [hunk 1]
```diff
@@ -103,5 +109,33 @@ def token2header(token_file):
...
+                            meta_host: str, ts_host: str, headers=None, session=None):
+    params = {'start': start_time.isoformat(),
+              'end': end_time.isoformat(),
+              'aggregate': aggregate}
+
+    rq = session or requests
+    if headers is None:
+        headers = {}
+    trace_id = str(uuid.uuid4())
+    headers['Trace-Id'] = trace_id
```
Comment (haakondr) [inline pyniva/get_data.py:126]: probably not something that we'd want to spend time on now, but all these headers should ideally be added as request middleware made by Johannes https://github.com/NIVANorge/nivacloud-logging#tracing-with-requests

File: tests/test_ms_ferrybox_nivatest.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+TSB_URL = 'https://api-test.niva.no/v1/tsb/'
+META_URL = 'https://api-test.niva.no/v1/metaflow/'
+
+
+@pytest.mark.token_and_ms_ferrybox
+def test_ms_ferrybox_for_newly_populated_data():
+    """ Testing that the ms ferrybox api responds correctly to requests for newly inserted data"""
+    header = pyniva.token2header("/home/roald/PycharmProjects/other_nivacloud_stuff/"
+                                 "secrets/nivatestdev-1b5ad93f6938.json")
```
Comment (haakondr) [inline tests/test_ms_ferrybox_nivatest.py:16]: these tests are good. I think that we should incorporate them into our CI to run on a scheduled interval on nivatest/prod

Comment (haakondr) [inline tests/test_ms_ferrybox_nivatest.py:16]: TODO: pass in token as env variable or path to file containing token. I think just passing in the token as env variable is easiest when setting up circleci secrets?

File: .circleci/config.yml [hunk 3]
```diff
@@ -28,6 +28,7 @@ jobs:
             python3 -m venv venv
             . venv/bin/activate
             pip install .[test]
+            pip install pytest
```
Comment (haakondr) [inline .circleci/config.yml:31]: should pytest not be installed as part of pip install ? need to somehow define it as a dependency in setup.py.

Side-note: I find setup.py to be quite confusing, I did experiment a bit with https://python-poetry.org/ and liked it better than using setup.py/setuptools. Have not really tested it in depth though

**Review by haakondr — APPROVED** [review]

File: tests/test_ms_ferrybox_nivatest.py (hunk 2 above)
Comment (ZofiaRudjord) [reply tests/test_ms_ferrybox_nivatest.py:16]: Is there a reason why you did not do that @RmStorm ?

File: .circleci/config.yml (hunk 3 above)
Comment (RmStorm) [reply .circleci/config.yml:31]: agreed on poetry! It looks great. It's not ideal with docker but good enough.

**Pull request merged**